### PR TITLE
Making sure `npm init ember-app` will always use the locally installed ember-cli instead of relying on the globally installed one

### PR DESCRIPTION
**Pull request by mansona**
I noticed something odd recently when using this in that it would always use the globally installed ember-cli (or the `ember` command) to create the new app. This is not quite what I expected 🤔 

This also creates a few issues in that if a user doesn't have `ember-cli` installed globally then this script will fail and not even show any error.

This PR adds tests on Travis that **don't** have ember-cli globally installed so it makes sure that it will work as expected on end-users' computers. 

I also removed the yarn.lock because this package will predominantly be used through `npm init` and `yarn create` and the yarn.lock was causing the Travis tests to pin the version of ember-cli instead of relying on the `^` semantics.

### CODE REVIEW

**Comment by Turbo87** [comment]
according to https://github.com/sindresorhus/execa/#preferlocal it should use the locally installed `ember-cli` command. do you think this is a bug in `execa`?

**Comment by mansona** [comment]
@Turbo87 I suppose this could be a bug in `execa` 🤔 but I'm wondering if there is a bit of a layer of indirection that doesn't quite work with how `execa` is implemented. I guess it's expecting to be run in a folder that has a `node_modules` already available and therefore it will be looking for `node_modules/.bin/ember` and when it doesn't find it then it will fall back to the global.

I guess the question in this context is "what does `local` mean" 🤔 What **we** need would be for it to refer to wherever `create-ember-app` is installed but I have a feeling that it instead is closer to `__dirname`

**Comment by Turbo87** [comment]
`execa` is using https://github.com/sindresorhus/npm-run-path under the hood, maybe that gives you a clue. it seems that `npx` is doing things in a slightly different way compared to `yarn create` :-/

**Comment by mansona** [comment]
@Turbo87 this issue exists the same in npm and yarn. If you uninstall ember globally and run `yarn create ember-app test-project` it will fail without telling you why. https://github.com/ember-cli/create-ember-app/pull/6/files#diff-168726dbe96b3ce427e7fedce31bb0bcR22 adds an error message in this case but this PR also fixes the underlying issue too 👍

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+language: node_js
+node_js:
+  - "8"
+
+sudo: false
+dist: trusty
```
Comment (Turbo87) [inline .travis.yml]: trusty is default now, and `sudo: false` is deprecated

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+branches:
+  only:
+    - master
+
+matrix:
+  include:
+  - name: "npm"
+    before_install:
+      - npm config set spin false
```
Comment (Turbo87) [inline .travis.yml]: is this still a thing with npm?! 😱

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    install:
+      - npm link
+    script:
+      - cd ../
+      - npm init ember-app test-app
+      - cd test-app
+      - "node_modules/.bin/ember -v | grep -q 'ember-cli: 3.6'"
```
Comment (Turbo87) [inline .travis.yml]: seems like this would fail as soon as 3.7 is released 🤔

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  - name: "Yarn"
+    install:
+      - yarn && yarn link
```
Comment (Turbo87) [inline .travis.yml]: can you make that two lines?

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  only:
+    - master
+
+matrix:
+  include:
+  - name: "npm"
+    before_install:
+      - npm config set spin false
+    install:
+      - npm link
```
Comment (Turbo87) [inline .travis.yml:14]: no `npm install` here?

File: .gitignore [hunk 6]
```diff
@@ -57,3 +57,6 @@ typings/
 # dotenv environment variables file
 .env
 
+# never commit lock files to keep testing working
+yarn.lock
+package-lock.json
```
Comment (Turbo87) [inline .gitignore:62]: why is this needed again?

File: .travis.yml (hunk 1 above)
Comment (mansona) [reply .travis.yml]: I used the current .travis.yml file from Ember as a prototype of this 🙈 https://github.com/ember-cli/ember-new-output/blob/stable/.travis.yml#L22 I assume we should be adding this feedback there too?

File: .travis.yml (hunk 2 above)
Comment (mansona) [reply .travis.yml]: tbh I think that Travis defaults to `npm ci` now so no 😂

File: .travis.yml (hunk 3 above)
Comment (mansona) [reply .travis.yml]: So I was thinking what I would want to do here to make sure this is working. I didn't want to just do `ember test` because that wouldn't verify that it was installing the latest ember 🤔 

I was thinking that I'm ok with this failing when 3.7 is released but I also wondered, is there a canonical "api" anywhere that shows the current latest ember version?

File: .travis.yml (hunk 5 above)
Comment (mansona) [reply .travis.yml:14]: npm link does an npm install at the same time 👍

File: .gitignore (hunk 6 above)
Comment (mansona) [reply .gitignore:62]: This is because of this failure: https://travis-ci.com/ember-cli/create-ember-app/jobs/167640273#L509

Essentially if you have a yarn.lock or a package-lock.json it will fail because the `npm install` or `yarn install` steps will be installing the locked version and not what is defined in the package.json (currently set to `"ember-cli": "^3.5.1"`) and therefore would not install the latest version

File: .travis.yml (hunk 1 above)
Comment (Turbo87) [reply .travis.yml]: yeah, I think a PR for that is still missing 🙈

File: .travis.yml (hunk 2 above)
Comment (Turbo87) [reply .travis.yml]: but we're not using that anywhere, are we?

File: .travis.yml (hunk 3 above)
Comment (Turbo87) [reply .travis.yml]: https://github.com/ember-cli/ember-source-channel-url maybe? not sure

File: .travis.yml (hunk 2 above)
Comment (mansona) [reply .travis.yml]: yes that's true, it's part of the `npm link`. It seems to work fine without needing this line 👍

File: .gitignore (hunk 6 above)
Comment (Turbo87) [reply .gitignore:62]: hmm... okay...

File: .travis.yml [hunk 7]
```diff
@@ -16,7 +16,8 @@ matrix:
       - cd ../
       - npm init ember-app test-app
       - cd test-app
-      - "node_modules/.bin/ember -v | grep -q 'ember-cli: 3.6'"
+      - "export CURRENT_VERSION=`curl -H 'Authorization: token $GITHUB_TOKEN' https://api.github.com/repos/ember-cli/ember-cli/releases/latest | jq '.tag_name' | grep -o '\\d\\+\\.\\d\\+\\.\\d\\+'`"
+      - "node_modules/.bin/ember -v | grep -q 'ember-cli: $CURRENT_VERSION'"
```
Comment (Turbo87) [inline .travis.yml]: this is starting to look quite complicated. would it be worth it to use a proper JS test runner to run those commands? I assume that might make it possible to run the tests locally too? 🤔

Comment (mansona) [reply .travis.yml]: If we want to run the ember tests we can literally just add the line `npm test` or `yarn test` to the end of this. My reasoning for **not** doing this was that running the ember tests doesn't really test the _point_ of this repo. All we really need to know is what version `ember -v` reports to us.

I'm also not 100% sure this is all that complicated compared to building test-runner infrastructure 🤔 I could maybe clean this "oneliner" up a bit to make it less complicated looking if that would help?

Comment (Turbo87) [reply .travis.yml]: I was not talking about running the ember tests. What I meant was writing a Mocha/Jest test suite that runs these commands and tests via npm and yarn as part of that test suite.

Comment (mansona) [reply .travis.yml]: But these tests need to be run in the context of the **newly created ember app**, not in the context of the project itself. That's why I didn't originally create a test script that could be run with `npm test` in the first place 😞 

I really don't think that these sorts of tests belong in a JS test runner because they are essentially **infrastructure** tests that want to test something that is as close as possible to running `npm init ember-app test-project` directly

Comment (Turbo87) [reply .travis.yml]: that's why I would use something like `execa` to run `npm init ember-app test-project` from within the test runner. basically, instead of using shell commands we use JS to do the same thing, and we wrap it in a test runner to get nicer feedback.

I might be over-engineering this though... 😅

Comment (mansona) [reply .travis.yml]: I wasn't really going for a "nice feedback" test runner scenario here 😖 all I really want is a single smoke test in both npm and yarn to make sure that when someone runs this they get what **we** expect them to get. 

For example, I tweeted about this functionality a while back and until I noticed that `npm init ember-app test-project` didn't automatically create an up-to-date project for me I had no clue that this was failing for everyone without ember installed globally 😭 

These "messy" tests are a nice-to-have because there was literally nothing there before. I am happy to do whatever to get this landed but I don't think adding a test runner in JS helps with the goal to make this "simpler"

Comment (Turbo87) [reply .travis.yml]: yeah, makes sense, we can always iterate on it if needed. having this as a basic smoke test is already much better than what we had before :)

**Review by Turbo87 — APPROVED** [review]
